JTS fails to import JSIDL with messages that don't have the 'is_command' attribute.

JTS fails to import messages that don't have the ```is_command``` attribute.

The ```is_command``` attribute is optional. However, JTS fails to import JSIDL with messages that don't have the ```is_command``` attribute. The import process simply reports "JTS fails to import messages that don't have the ```is_command``` attribute."

It looks like a NullPointerException is getting thrown:
```
Caused by: java.lang.NullPointerException
	at org.jts.gui.JAXBtoJmatter.MessageDef.lookupOrCreate(MessageDef.java:80)
	at org.jts.gui.JAXBtoJmatter.InputSet.lookupOrCreate(InputSet.java:84)
	at org.jts.gui.JAXBtoJmatter.ServiceDef.createServiceDef(ServiceDef.java:168)
	at org.jts.gui.JAXBtoJmatter.ServiceDef.lookupOrCreate(ServiceDef.java:65)
	at org.jts.gui.importJSIDL.Import.commitInOrder(Import.java:829)
	at org.jts.gui.importJSIDL.Import.access$700(Import.java:79)
	at org.jts.gui.importJSIDL.Import$ImportDefsWorker.performImport(Import.java:646)
	at org.jts.gui.importJSIDL.Import$ImportDefsWorker.run(Import.java:255)
	... 1 more
```

--- a/GUI/src/org/jts/codegenerator/MessageDefGenerator.java
+++ b/GUI/src/org/jts/codegenerator/MessageDefGenerator.java
@@ -129,7 +129,7 @@ public class MessageDefGenerator {
                 }
 
                 /// Initialize isCommand variable
-                code.constructorLines.add("m_IsCommand = " + String.valueOf(msgDef.isIsCommand()) + ";");
+                code.constructorLines.add("m_IsCommand = " + String.valueOf(msgDef.isIsCommand() == null ? false : msgDef.isIsCommand()) + ";");
 
                 includes.add("Messages/Message.h");
                 includes.add("JConstants.h");
@@ -202,7 +202,7 @@ public class MessageDefGenerator {
                     code.addAll(subCode);
                 }
                 /// Initialize isCommand variable
-                code.constructorLines.add("m_IsCommand = " + String.valueOf(msgDef.isIsCommand()) + ";");
+                code.constructorLines.add("m_IsCommand = " + String.valueOf(msgDef.isIsCommand() == null ? false : msgDef.isIsCommand()) + ";");
 
                 includes.add("framework.messages.Message");
                 includes.add("framework.JausUtils");
@@ -298,7 +298,7 @@ public class MessageDefGenerator {
                     code.addAll(subCode);
                 }
                 /// Initialize isCommand variable
-                code.constructorLines.add("m_IsCommand = " + String.valueOf(msgDef.isIsCommand()) + ";");
+                code.constructorLines.add("m_IsCommand = " + String.valueOf(msgDef.isIsCommand() == null ? false : msgDef.isIsCommand()) + ";");
 
                 baseClassList.add("JTS.Message");
                 if (!namespace.isEmpty()) {

--- a/GUI/src/org/jts/gui/JAXBtoJmatter/MessageDef.java
+++ b/GUI/src/org/jts/gui/JAXBtoJmatter/MessageDef.java
@@ -77,7 +77,9 @@ public class MessageDef
 			jmMessageDef.getMessageId().setValue(messageId);
 			
 			// isCommand
-			jmMessageDef.getIsCommand().setValue(jxMessageDef.isIsCommand());
+                        /// @TODO The isCommand attribute is optional. However, 
+                        /// the jmMessage does not support optional booleans. 
+			jmMessageDef.getIsCommand().setValue(jxMessageDef.isIsCommand() == null ? false : jxMessageDef.isIsCommand());
 			
 			// Description
 			String description = jxMessageDef.getDescription().getContent().replaceAll("\\s\\s+ | \\n | \\r | \\t", " ").trim();;			

--- a/GUI/src/org/jts/gui/jmatterToJAXB/MessageDef.java
+++ b/GUI/src/org/jts/gui/jmatterToJAXB/MessageDef.java
@@ -45,6 +45,7 @@ public class MessageDef {
     md.setName( messageDef.getName().toString() );
      
      md.setMessageId( getBytes(messageDef.getMessageId().toString()) );
+     /// @TODO: This should handle null values because the isCommand is optional.
      md.setIsCommand( messageDef.getIsCommand().booleanValue() );
      md.setDescription( Description.convert( messageDef) );
      

--- a/promelaCodeGenerator/src/org/jts/protocolvalidator/DefinitionGenerator.java
+++ b/promelaCodeGenerator/src/org/jts/protocolvalidator/DefinitionGenerator.java
@@ -136,7 +136,7 @@ public class DefinitionGenerator extends JTSFileWriter {
         // get type from reference to the type
         String name = Util.getTypeNameFromDeclTypeRef(refMap, msgDef.getName(), classID);
         refMap.put(msgDef.getName(), name);
-        boolean cmd = msgDef.isIsCommand();
+        boolean cmd = msgDef.isIsCommand() == null ? false : msgDef.isIsCommand();
         waitingOutput.add("typedef " + name + "{");
 
         // definitions and declarations need to be kept separate

